### DISCUSSION TITLE
Looks like YouTube wants login

### DISCUSSION THREAD

**Discussion by HagarJNode** [lavalink-devs/youtube-source#72] [Q&A] [answered]
I'm getting a FriendlyException using version 1.8.3 - have been using it for a while now without any problems.

`20:43:16.040 WARN  c.s.d.l.t.p.LocalAudioTrackExecutor - Suspicious exception for playback of DrQRS40OKNE
com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Please sign in
        at dev.lavalink.youtube.track.YoutubeAudioTrack.process(YoutubeAudioTrack.java:102)
        at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:109)
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$executeTrack$2(DefaultAudioPlayerManager.java:339)
`

Any ideas on what to do?

**Comment by devoxin** [comment, marked as answer]
use OAuth.

**Reply by HagarJNode** [reply]
From a Discord bot??

**Reply by devoxin** [reply]
Yes?

**Reply by HagarJNode** [reply]
Will try it out

**Reply by HagarJNode** [reply]
Didn't work...

```
00:00:41.411 INFO  d.l.y.http.YoutubeOauth2Handler - ==================================================
00:00:41.412 INFO  d.l.y.http.YoutubeOauth2Handler - !!! DO NOT AUTHORISE WITH YOUR MAIN ACCOUNT, USE A BURNER !!!
00:00:41.412 INFO  d.l.y.http.YoutubeOauth2Handler - OAUTH INTEGRATION: To give youtube-source access to your account, go to https://www.google.com/device and enter code SJY-LZV-FJD
00:00:41.412 INFO  d.l.y.http.YoutubeOauth2Handler - !!! DO NOT AUTHORISE WITH YOUR MAIN ACCOUNT, USE A BURNER !!!
00:00:41.413 INFO  d.l.y.http.YoutubeOauth2Handler - ==================================================
00:00:55.906 INFO  c.s.l.c.natives.NativeLibraryLoader - Native library libmpg123-0: loading with filter com.sedmelluq.discord.lavaplayer.natives.ConnectorNativeLibLoader$$Lambda$384/0x00007f47e57c6180@23a4826c
00:00:55.916 INFO  c.s.l.c.natives.NativeLibraryLoader - Native library connector: loading with filter null
00:00:56.020 INFO  c.s.l.c.natives.NativeLibraryLoader - Native library connector: successfully loaded.
00:04:57.240 INFO  d.l.y.http.YoutubeAccessTokenTracker - Updating YouTube visitor id succeeded, new one is CgtidDJCWGRxdnAtNCjlrPC4BjIiCgJTRRIcEhgSFgsMDg8QERITFBUWFxgZGhscHR4fICEgFzoMCAEg57mnvtPMhY5n, next update will be after 600 seconds.
00:04:57.334 WARN  c.s.d.l.t.p.LocalAudioTrackExecutor - Suspicious exception for playback of vYaimKb6FH0
com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Please sign in
        at dev.lavalink.youtube.track.YoutubeAudioTrack.process(YoutubeAudioTrack.java:102)
        at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:109)
        at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$executeTrack$2(DefaultAudioPlayerManager.java:339)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
00:05:27.235 WARN  c.s.d.l.t.p.LocalAudioTrackExecutor - Suspicious exception for playback of N4pd4frmGW4
com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Please sign in
        at dev.lavalink.youtube.track.YoutubeAudioTrack.process(YoutubeAudioTrack.java:102)
```

**Reply by HagarJNode** [reply]
Works - thanks.

**Reply by JReddondo** [reply]
Hello @HagarJNode!

I'm having trouble configuring YoutubeAudioSourceManager with OAuth2. I've generated a token and passed it to the useOauth2 method as shown below, which should trigger the if statement to set up the manager. However, when I run the application, I see the expected log message indicating that the token is defined, but I still receive a FriendlyException prompting me to log in. Can you help me resolve this issue? I'm using youtube-source with Lavaplayer, am I missing some yaml configuration or perhaps I require Lavalink too?

Logs:
> Youtube token defined
> YouTube access token refreshed successfully
> Login Successful!

Code:
```
if (!StringUtils.isEmpty(youtubeProperties.getToken())) {
	logger.info("Youtube token defined");
	audioSourceManager.useOauth2(youtubeProperties.getToken(), true);
} else {
	logger.info("No youtube token found");
	audioSourceManager.useOauth2(null, false);
}
```

**Reply by devoxin** [reply]
Are you using the TV clients?

**Reply by JReddondo** [reply]
I guess, I'm not sure. I have copied the code in the readme because if I use the empty constructor I get the exception below for some videos

`YoutubeAudioSourceManager audioSourceManager = new YoutubeAudioSourceManager(true, new MusicWithThumbnail(), new WebWithThumbnail(), new AndroidMusicWithThumbnail());`

> **com.sedmelluq.discord.lavaplayer.tools.FriendlyException: Loading information for video failed**
> 	at dev.lavalink.youtube.track.YoutubeAudioTrack.process(YoutubeAudioTrack.java:114)
> 	at com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor.execute(LocalAudioTrackExecutor.java:109)
> 	at com.sedmelluq.discord.lavaplayer.player.DefaultAudioPlayerManager.lambda$executeTrack$2(DefaultAudioPlayerManager.java:339)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
> 	at java.base/java.lang.Thread.run(Thread.java:1583)
> **Caused by: java.lang.RuntimeException: Non-embeddable video cannot be loaded by embedded client**
> 	at dev.lavalink.youtube.clients.skeleton.NonMusicClient.loadTrackInfoFromInnertube(NonMusicClient.java:151)
> 	at dev.lavalink.youtube.clients.skeleton.StreamingNonMusicClient.loadFormats(StreamingNonMusicClient.java:35)
> 	at dev.lavalink.youtube.track.YoutubeAudioTrack.loadBestFormatWithUrl(YoutubeAudioTrack.java:194)
> 	at dev.lavalink.youtube.track.YoutubeAudioTrack.processWithClient(YoutubeAudioTrack.java:131)
> 	at dev.lavalink.youtube.track.YoutubeAudioTrack.process(YoutubeAudioTrack.java:79)

**Reply by devoxin** [reply]
You're not using any TV clients.

**Reply by JReddondo** [reply]
I just included every client and it's working. Thanks for your time!

**Comment by WeeeeeeeeeeS** [comment]
Do you have eyes? Use them to read the first five lines💀

**Reply by HagarJNode** [reply]
I read those... not that practical for discord bots, or is that only me that see it like that?

**Reply by devoxin** [reply]
Then you should consider reading the entirety of the "Using OAuth Tokens" section in the README which tells you specifically that once you complete the OAuth flow at least **once**, you will get a refresh token that you can use from thereon, thus avoiding the need to do that flow again.

**Reply by HagarJNode** [reply]
Will read that part again doing the weekend and see if I can find a way this oauth...

**Comment by devoxin** [comment]
If you are not prepared to use OAuth then there is nothing more to be gained from this discussion. OAuth is the only practical solution we have at the moment as YouTube are cracking down more on bots using their internal API, and more broadly, streaming from their service. There is no other (publicly known) solution at this point in time.

**Reply by HagarJNode** [reply]
Well, I hope to find a solution, but I guess loads of others will be here asking the same question in a while...

**Reply by devoxin** [reply]
Again, the solution is OAuth. And everyone else is using it.

**Reply by LucasB25** [reply]
> If you are not prepared to use OAuth then there is nothing more to be gained from this discussion. OAuth is the only practical solution we have at the moment as YouTube are cracking down more on bots using their internal API, and more broadly, streaming from their service. There is no other (publicly known) solution at this point in time.

the potoken system too?

**Reply by devoxin** [reply]
Unreliable. Youtube has mostly patched it

**Reply by LucasB25** [reply]
I confirm that it does not work with the potoken
